### PR TITLE
nit: Cleanup staff and superuser

### PR DESCRIPTION
**Pull request by schew2381**
Addressing [comments](https://github.com/getsentry/sentry/pull/61653#pullrequestreview-1789762901)

Did some of the easier ones.
Applied all changes to superuser as well except for allowing the setting of `current_datetime` in Superuser `__init__`

### CODE REVIEW

File: tests/sentry/auth/test_staff.py [hunk 1]
```diff
@@ -182,10 +182,11 @@ def test_login_saves_session(self):
         request.user = user
         assert staff.is_active
 
-        data = request.session.get(SESSION_KEY)
+        # See mypy issue: https://github.com/python/mypy/issues/9457
+        data = request.session.get(SESSION_KEY)  # type:ignore[unreachable]
```
Comment (schew2381) [inline tests/sentry/auth/test_staff.py:186]: Annoying bug with mypy not assuming new assert value could change, guido responded to issue tho which is cool
https://github.com/python/mypy/issues/9457

File: src/sentry/auth/staff.py [hunk 2]
```diff
@@ -31,9 +32,10 @@
 COOKIE_HTTPONLY = getattr(settings, "STAFF_COOKIE_HTTPONLY", True)
 
 # the maximum time the session can stay alive
-MAX_STAFF_SESSION_AGE = timedelta(hours=4)
+MAX_AGE = timedelta(hours=4)
```
Comment (schew2381) [inline src/sentry/auth/staff.py:35]: Changing this b/c this is the standard constant name in some of the package cookie functions

File: src/sentry/auth/elevated_mode.py [hunk 3]
```diff
@@ -1,38 +1,47 @@
...
+
+
+class InactiveReason(str, Enum):
+    INVALID_IP = "invalid-ip"
+    INCOMPLETE_SSO = "incomplete-sso"
+    # Indicates the request should be allowed
+    NONE = None
+
+    def __bool__(self):
+        return self.value is not None
```
Comment (schew2381) [inline src/sentry/auth/elevated_mode.py:13]: We use the enum in a truthy check, InactiveReason.NONE should be falsey

File: src/sentry/auth/elevated_mode.py [hunk 4]
```diff
@@ -1,38 +1,47 @@
...
 
-    @classmethod
     @abstractmethod
-    def is_privileged_request(self):
+    def is_privileged_request(self) -> Tuple[bool, InactiveReason]:
         pass
 
-    @classmethod
     @abstractmethod
     def get_session_data(self, current_datetime=None):
```
Comment (cathteng) [inline src/sentry/auth/elevated_mode.py:30]: i think we can add a return type of `None` here

File: src/sentry/auth/elevated_mode.py [hunk 5]
```diff
@@ -1,38 +1,47 @@
...
         pass
 
-    @classmethod
     @abstractmethod
     def _populate(self, current_datetime=None):
```
Comment (cathteng) [inline src/sentry/auth/elevated_mode.py:34]: return `None` here too?

File: src/sentry/auth/elevated_mode.py (hunk 3 above)
Comment (cathteng) [reply src/sentry/auth/elevated_mode.py:13]: here right? https://github.com/getsentry/sentry/blob/f906a6cc4686609360aba7ab83b0947e7ac8c3e2/src/sentry/auth/superuser.py#L282

Comment (schew2381) [reply src/sentry/auth/elevated_mode.py:13]: Yup that's the one

File: src/sentry/auth/elevated_mode.py (hunk 4 above)
Comment (schew2381) [reply src/sentry/auth/elevated_mode.py:30]: session_data returns a JSON I think at the end, so I left it untyped for now :|

Comment (cathteng) [reply src/sentry/auth/elevated_mode.py:30]: i see :/

File: src/sentry/auth/superuser.py [hunk 6]
```diff
@@ -320,16 +322,21 @@ def _set_logged_in(self, expires, token, user, current_datetime=None):
...
         self.is_valid = False
         self.request.session.pop(SESSION_KEY, None)
 
-    def set_logged_in(self, user, prefilled_su_modal=None, current_datetime=None):
+    def set_logged_in(
+        self,
+        user,
+        current_datetime=None,
+        prefilled_su_modal=None,
+    ) -> None:
```
Comment (schew2381) [inline src/sentry/auth/superuser.py:339]: Have to switch the order of arguments to comply with the interface

File: src/sentry/auth/elevated_mode.py [hunk 7]
```diff
@@ -1,38 +1,47 @@
 from abc import ABC, abstractmethod
+from enum import Enum
+from typing import Tuple
+
+
+class InactiveReason(str, Enum):
+    INVALID_IP = "invalid-ip"
+    INCOMPLETE_SSO = "incomplete-sso"
```
Comment (ykamo001) [inline src/sentry/auth/elevated_mode.py:8]: Should the string values be consistent with `_` as well, or is `-` the pattern we have to stick to?

File: src/sentry/auth/staff.py [hunk 8]
```diff
@@ -54,17 +56,17 @@ def is_active_staff(request: Request) -> bool:
 class Staff(ElevatedMode):
     allowed_ips = frozenset(ipaddress.ip_network(str(v), strict=False) for v in ALLOWED_IPS)
 
-    def __init__(self, request, allowed_ips=UNSET, current_datetime=None):
+    def __init__(self, request, allowed_ips=UNSET):
```
Comment (ykamo001) [inline src/sentry/auth/staff.py:59]: nit
```suggestion
    def __init__(self, request, allowed_ips=UNSET) -> None:
```

File: src/sentry/auth/elevated_mode.py (hunk 7 above)
Comment (ykamo001) [reply src/sentry/auth/elevated_mode.py:8]: Answered this myself, this is ported over from the existing values as is, so might be needed with the hyphen/dash

File: src/sentry/auth/elevated_mode.py [hunk 9]
```diff
@@ -1,38 +1,47 @@
...
-    @classmethod
     @abstractmethod
-    def _populate(self, current_datetime=None):
+    def _populate(self) -> None:
         pass
 
-    @classmethod
     @abstractmethod
-    def set_logged_in(self, user, current_datetime=None):
+    def set_logged_in(self, user, current_datetime=None) -> None:
```
Comment (ykamo001) [inline src/sentry/auth/elevated_mode.py:38]: do we need the current time for this?

**Review by ykamo001 — APPROVED** [review]

File: src/sentry/auth/elevated_mode.py (hunk 9 above)
Comment (schew2381) [reply src/sentry/auth/elevated_mode.py:38]: It's used for testing in superuser which I copied over to staff

**Pull request merged**